src/lib.rs

```diff
@@ -176,6 +176,62 @@ impl<F: Float, C: FloatChecker<F>> NoisyFloat<F, C> {
         }
     }
 
+    /// Converts the value in-place to a reference to a `NoisyFloat`.
+    ///
+    /// Uses the `FloatChecker` to assert that the value is valid.
+    #[inline]
+    pub fn borrowed(value: &F) -> &Self {
```

Comment:
Since borrowed(...) relies on a tricky unsafe cast, you should add a unit test to make sure it works as intended.

Reply:
I've added a few lines of tests as a sanity check. Are they sufficient?